src/dos/cdrom.cpp

```diff
@@ -222,17 +238,46 @@ bool CDROM_Interface_Fake :: GetAudioSub(unsigned char& attr, unsigned char& tra
 	track	= index = 1;
 	relPos.min = relPos.fr = 0; relPos.sec = 2;
 	absPos.min = absPos.fr = 0; absPos.sec = 2;
+
+	LagDriveResponse();
 	return true;
 }
 
 bool CDROM_Interface_Fake :: GetAudioStatus(bool& playing, bool& pause) {
 	playing = pause = false;
+
+	LagDriveResponse();
 	return true;
 }
 
 bool CDROM_Interface_Fake :: GetMediaTrayStatus(bool& mediaPresent, bool& mediaChanged, bool& trayOpen) {
 	mediaPresent = true;
 	mediaChanged = false;
 	trayOpen     = false;
+
+	LagDriveResponse();
 	return true;
 }
+
+// Simulate the delay a physical CD-ROM drive took to respond to queries. When
+// added to calls, this ensures that back-to-back queries report monotonically
+// increasing Minute-Second-Frame (MSF) time values.
+//
+void CDROM_Interface::LagDriveResponse() const
+{
+	// Always simulate a very small amount of drive response time
+	CALLBACK_Idle();
+
+	// Handle tick-rollover
+	static decltype(PIC_Ticks) prev_ticks = 0;
```

Comment:
Why not just `static PIC_Ticks prev_ticks = 0;`? What does this give us?

Reply:
(it looks like a type, but `PIC_Ticks` is the variable holds the tick count) - so in this case, I'm asking for whatever type `PIC_Ticks` is, that way if we change it, this follows suite).